src/bauhaus/bauhaus.c

```diff
@@ -426,32 +384,54 @@ static gboolean dt_bauhaus_popup_scroll(GtkWidget *widget, GdkEventScroll *event
 
 static gboolean dt_bauhaus_popup_motion_notify(GtkWidget *widget, GdkEventMotion *event, gpointer user_data)
 {
-  GtkAllocation allocation_popup_window;
-  gtk_widget_get_allocation(darktable.bauhaus->popup_window, &allocation_popup_window);
   gtk_widget_queue_draw(darktable.bauhaus->popup_area);
-  dt_bauhaus_widget_t *w = darktable.bauhaus->current;
-  GtkAllocation allocation_w;
-  gtk_widget_get_allocation(GTK_WIDGET(w), &allocation_w);
-  const GtkBorder *padding = darktable.bauhaus->popup_padding;
-  const int width = allocation_popup_window.width - padding->left - padding->right;
-  const int height = allocation_popup_window.height - padding->top - padding->bottom;
-  const int ht = darktable.bauhaus->line_height + INNER_PADDING * 2.0f;
-  // coordinate transform is in vain because we're only ever called after a button release.
-  // that means the system is always the one of the popup.
-  // that also means that we can't have hovering combobox entries while still holding the button. :(
-  const float ex = event->x - padding->left;
-  const float ey = event->y - padding->top;
+
   GtkAllocation allocation;
   gtk_widget_get_allocation(widget, &allocation);
 
+  const GtkBorder *padding = darktable.bauhaus->popup_padding;
+  const int width = allocation.width - padding->left - padding->right;
+  const int height = allocation.height - padding->top - padding->bottom;
+  const int ht = darktable.bauhaus->line_height + INNER_PADDING * 2.0f;
+
+  gint wx, wy;
+  GdkWindow *window = gtk_widget_get_window(darktable.bauhaus->popup_window);
+  gdk_window_get_origin(window, &wx, &wy);
+
+  const float tol = 50;
+  if(event->x_root > wx + allocation.width + tol || event->y_root > wy + allocation.height + tol
+     || event->x_root < (int)wx - tol || event->y_root < (int)wy - tol)
+  {
+    dt_bauhaus_widget_reject(darktable.bauhaus->current);
+    dt_bauhaus_hide_popup();
+    return TRUE;
+  }
+
+  const float ex = event->x_root - wx - padding->left;
+  const float ey = event->y_root - wy - padding->top;
+
   if(darktable.bauhaus->keys_cnt == 0) _stop_cursor();
 
+  dt_bauhaus_widget_t *w = darktable.bauhaus->current;
   switch(w->type)
   {
     case DT_BAUHAUS_COMBOBOX:
+    {
+      GdkRectangle workarea;
+      gdk_monitor_get_workarea(gdk_display_get_monitor_at_window(gdk_window_get_display(window), window), &workarea);
+
+      gint dy = 0, move = darktable.bauhaus->mouse_y - ey;
+      if(ey < darktable.bauhaus->mouse_y && wy < workarea.y)
+        dy = (workarea.y - wy) * move / (darktable.bauhaus->mouse_y + wy - workarea.y);
```

Comment:
if I understand, you want that when your cursor reach the top of the main widow, the popup top reach the same point, right ?
In this case, that means that you want that a move of the cursor of `(darktable.bauhaus->mouse_y - workarea.y)` do a move of `(workarea.y - wy)` For a smaller movement of `move`, that would mean a popup movement of `dy = (workarea.y - wy) * move / (darktable.bauhaus->mouse_y - workarea.y);` (note the absence of the `wy` in the second part...

Reply:
I want to compare the previous mouse position on the _screen_ to the edge. mouse_y is relative to the popup, which at this point is still a little off the screen (and therefore the mouse can't reach it).